src/http/common.cr

```diff
@@ -139,6 +139,13 @@ module HTTP
   end
 
   def self.serialize_headers_and_string_body(io, headers, body)
+    if headers.includes_word?("Content-Encoding", "gzip")
```

Comment:
We should compare with `==` instead of `includes_word?` here to match what we do when decompressing (we only accept "gzip" exactly). This is because `Content-Encoding` can represent multiple encodings applied in sequence. We only want to support this when the only encoding is gzip - not just when gzip is part of the encoding sequence. THis is very rare in practice, but should be in consideration.

Reply:
Nice! You are right. I'll fix it. Thanks.